src/Pagination/Pagination.test.jsx

```diff
@@ -0,0 +1,257 @@
+      wrapper.find('.previous').simulate('click');
+      expect(spy).toHaveBeenCalledTimes(1);
+    });
+
+    it('next button onClick', () => {
+      const spy = jest.fn();
+      const props = {
+        ...baseProps,
+        onPageSelect: spy,
+      };
+      const wrapper = mount(<Pagination {...props} />);
+      wrapper.find('.next').simulate('click');
+      expect(spy).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('renders button label props', () => {
+    const buttonLabels = {
+      previous: 'Anterior',
+      next: 'Siguiente',
+      page: 'Página',
+      currentPage: 'Página actual',
+      pageOfCount: 'de',
+    };
+
+    let wrapper;
+    let props;
+
+    beforeEach(() => {
+      // Reset window size to extra large display
+      global.innerWidth = breakpoints.extraLarge.minWidth;
+
+      props = {
+        ...baseProps,
+        buttonLabels,
+      };
+      wrapper = mount(<Pagination {...props} />);
+    });
+
+    it('uses passed in previous button label', () => {
+      expect(wrapper.findWhere(node => (
+        node.name() === 'button' && node.hasClass('previous')
+      )).prop('aria-label')).toEqual(buttonLabels.previous);
+
+      wrapper.setProps({ currentPage: baseProps.pageCount });
+
+      expect(wrapper.findWhere(node => (
+        node.name() === 'button' && node.hasClass('previous')
+      )).prop('aria-label')).toEqual(`${buttonLabels.previous}, ${buttonLabels.page} 4`);
+    });
+
+    it('uses passed in next button label', () => {
+      expect(wrapper.findWhere(node => (
+        node.name() === 'button' && node.hasClass('next')
+      )).prop('aria-label')).toEqual(`${buttonLabels.next}, ${buttonLabels.page} 2`);
+
+      wrapper.setProps({ currentPage: baseProps.pageCount });
+
+      expect(wrapper.findWhere(node => (
+        node.name() === 'button' && node.hasClass('next')
+      )).prop('aria-label')).toEqual(buttonLabels.next);
+    });
+
+    it('uses passed in page button label', () => {
+      expect(wrapper.state('currentPage')).toEqual(1);
+      expect(wrapper.find('.btn').at(1).prop('aria-label'))
+        .toEqual(`${buttonLabels.page} 1, ${buttonLabels.currentPage}`);
+
+      wrapper.setProps({ currentPage: 2 });
+      expect(wrapper.state('currentPage')).toEqual(2);
+
+      expect(wrapper.find('.btn').at(1).prop('aria-label'))
+        .toEqual(`${buttonLabels.page} 1`);
+
+      // Reset window size to extra small display
+      global.innerWidth = breakpoints.extraSmall.maxWidth;
+      wrapper = mount(<Pagination {...props} />);
+
+      expect(wrapper.find('.btn').at(1).prop('aria-label'))
+        .toEqual(`${buttonLabels.page} 1, ${buttonLabels.currentPage}, ${buttonLabels.pageOfCount} 5`);
+
+      // Reset window size back to extra large display
+      global.innerWidth = breakpoints.extraLarge.minWidth;
+    });
+  });
+
+  describe('maxPagesDisplayed prop', () => {
+    it('should throw error for non-valid value', () => {
```

Comment:
Do we want to test this? Is the goal just to make sure that the propTypes are what we expect them to be? Because otherwise I don't know that we should be testing that the propTypes package correctly logs warnings for propTypes that fail the validators; they should be doing that. Could we just import the propTypes object and validate that, if that's what you're doing here?

Reply:
I included this test only for code coverage on the custom Prop type validation function. But because that's going to get swapped out with the `between` validator from `airbnb-proptypes`, this test will be removed.

Reply:
Oh, I see.